src/main/java/Deadline.java

```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString().substring(3) + " (by: " + by + ")";
```

Comment:
Perhaps you can use some form of [" + super.getStatusIcon() + "] instead

Reply:
The aim of using "substring(3)" is to replace the default "[ ]" in task, already replaced by constant. Thx!